src/components/app.jsx

```diff
@@ -106,79 +32,89 @@ export default class App extends Component {
 		// This is to capture clicks outside of the main window and close if necessary while also
 		// whitelisting the 'showConsentTool' button
 		if (!config.blockBrowsing) {
-			document.addEventListener('click', function(event) {
-				const target = event.target;
-				const showConsentToolButtonClicked = RegExp('showConsentTool').test(target.getAttribute('onclick'));
-				const appDiv = self.base;
-				const { layout } = config;
-				const { isConsentToolShowing, isFooterConsentToolShowing, isThinConsentToolShowing, isFooterShowing } = store;
-
-				if ( (isConsentToolShowing || isFooterConsentToolShowing || isThinConsentToolShowing || isFooterShowing) &&
-					!showConsentToolButtonClicked &&
-					!appDiv.contains(target) ) {
-					store.toggleConsentToolShowing(false);
-
-					// Render footer style CMP if no consent decision has been submitted yet
-					if (!cmp.submitted) {
-						layout !== 'thin' ? store.toggleFooterConsentToolShowing(true) : store.toggleThinConsentToolShowing(true);
+			document.addEventListener(
+				"click",
+				event => {
+					const target = event.target;
+					const showConsentToolButtonClicked = RegExp("showConsentTool").test(
+						target.getAttribute("onclick")
+					);
+					const appDiv = self.base;
+					const { layout } = config;
+					const {
+						isConsentToolShowing,
+						isFooterConsentToolShowing,
+						isThinConsentToolShowing,
+						isFooterShowing
+					} = store;
+					const isPopupVisible =
+						isConsentToolShowing ||
+						isFooterConsentToolShowing ||
+						isThinConsentToolShowing ||
+						isFooterShowing;
+
+					if (
+						isPopupVisible &&
+						!showConsentToolButtonClicked &&
+						!appDiv.contains(target)
+					) {
+						store.toggleConsentToolShowing(false);
+
+						// Render footer style CMP if no consent decision has been submitted yet
+						if (!cmp.submitted) {
+							layout !== "thin"
+								? store.toggleFooterConsentToolShowing(true)
+								: store.toggleThinConsentToolShowing(true);
+						}
 					}
-				};
-			}, false);
+				},
+				false
+			);
 		}
 	}
 
-	componentDidMount() {
-		const { store } = this.state;
-		const { config } = this.props;
-
-		if (config.css["custom-font-url"]) {
-			let head = document.head;
-			let link = document.createElement("link");
-			link.type = "text/css";
-			link.rel = "stylesheet";
-			link.href = config.css["custom-font-url"];
-			head.appendChild(link);
-		}
+	onFooterClose = () => {
+		this.props.store.toggleFooterShowing(false);
+	};
 
-		store.subscribe(this.updateCSSPrefs);
-		this.updateCSSPrefs();
-	}
+	onShowConsent = () => {
+		this.props.store.cmp.commands.showConsentTool();
+	};
 
 	render(props, state) {
 		const { store } = state;
 		const { config } = props;
-		const userLocalization = config.localization[currentLocale.split('-')[0]];
+		const showPopup =
+			store.isConsentToolShowing ||
+			store.isFooterConsentToolShowing ||
+			store.isThinConsentToolShowing;
```

Comment:
@LassaLek When logging `isThinConsentToolShowing`, it doesn't appear to exist on the store object.